libs/labelbox/src/labelbox/adv_client.py

```diff
@@ -59,7 +60,18 @@ def _request(self,
                                         url,
                                         data=requests_data,
                                         headers=headers)
-        response.raise_for_status()
+        if response.status_code != 200:
+            response_json = response.json()
+            if response_json.get('status') == 'CONFLICT':
+                raise ADVError("A custom embedding with this name already exists. Custom embeddings must have unique names.")
+            message = response_json.get('message')
+            if message:
+                if message == "must be greater than or equal to 8":
+                    raise ADVError("Custom embeddings require at least 8 dimensions.")
+                else:
+                    raise ADVError(response.json()['message'])
+            else:
+                response.raise_for_status()
```

Comment:
I find this approach brittle for a couple reasons: 1) `_request` is a generic function and no assumption should be made wrt the calling function based on the value of `status` and 2) translating exception messages unnecessarily couples this implementation to the backend service, if we decide to change the exception message on the backend this will break - if you are not happy with the received message we should update on the backend.

Reply:
I hate parsing error messages too, I just didn't want to have a broader impact than what the ticket specified. I would much prefer to change the messages on ADV side to these. If you don't mind me doing that I'll remove these changes and edit the appropriate errors in ADV, and parse neither 'status' nor 'message' here.

Reply:
Sounds good, I can also work with you to update on the ADV side too, let me know.

At the end of the day the exception message in ADV are meant to be human-readable and if they are not we should update them.